pilot/pkg/xds/internalgen.go

```diff
@@ -52,12 +54,28 @@ const (
 type InternalGen struct {
 	Server *DiscoveryServer
 
-	Store model.ConfigStore
+	// Store should either be k8s (for running pilot) or in-memory (for tests). MCP and other config store implementations
+	// do not support writing.
+	Store model.ConfigStoreCache
+
+	// cleanupLimit rate limit's autoregistered WorkloadEntry cleanup
+	cleanupLimit *rate.Limiter
+
+	cleanupQueue queue.Delayed
 
 	// TODO: track last N Nacks and connection events, with 'version' based on timestamp.
 	// On new connect, use version to send recent events since last update.
 }
 
+func NewInternalGen(s *DiscoveryServer) *InternalGen {
+	return &InternalGen{
+		Server: s,
+		// TODO make this configurable
+		cleanupLimit: rate.NewLimiter(rate.Limit(20), 1),
+		cleanupQueue: queue.NewDelayed(queue.DelayQueueWorkers(10)),
```

Comment:
do we need or want 10 workers? seems like potentially quite a bit. But we probably won't know good settings here until we test in a live large scale cluster

Reply:
I guess not. An issue with the delayed queue impl is that when the buffer is full, we pause processing until a worker picks something up. I'll change this to be a lock, so we don't rely on buffered chans to avoid blocking sends.

Reply:
Updated the delay queue to use the channel to update the underlying heap, and fall back to locking the queue and pushing to the heap directly if the buffered chan is full.